### PR TITLE
Add TempoDebris Module with Basic Debris Graph

### PR DESCRIPTION
**Pull request by pete-tempo**
Adds a PCG graph for generating random debris, including optional clustering and "keepout" zones from one or more splines. Requires two additional PCG elements: randomly choosing a mesh distribution from a data table row, and copying to each point an attribute from the nearest point in another set of points.

### CODE REVIEW

File: TempoWorld/Source/TempoDebris/Public/PCGCopyAttributeFromNearest.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+public:
+	/** The attribute to copy from the input. */
+	UPROPERTY(BlueprintReadWrite, EditAnywhere, Category = Settings)
+	FPCGAttributePropertySelector InputAttribute = FPCGAttributePropertySelector::CreateAttributeSelector(TEXT("$Density"));
+
+	/** The attribute to store the copy in the output. */
+	UPROPERTY(BlueprintReadWrite, EditAnywhere, Category = Settings)
+	FPCGAttributePropertySelector OutputAttribute = FPCGAttributePropertySelector::CreateAttributeSelector(TEXT("$Density"));
+	
+    /** A maximum distance to search, which is used as an optimization */
```
Comment (pete-tempo) [inline TempoWorld/Source/TempoDebris/Public/PCGCopyAttributeFromNearest.h:42]: Tabs/spaces?

File: TempoWorld/Source/TempoDebris/Public/PCGAssignRandomMeshAttribute.h [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+{
+	GENERATED_BODY()
+
+	// The static meshes and associated relative probabilities to use.
+	UPROPERTY(EditAnywhere, BlueprintReadWrite)
+	TSet<FStaticMeshProportion> StaticMeshDistribution;
+};
+
+/**
+ * Assigns a random mesh from the specified data table row to the specified attribute name.
```
Comment (pete-tempo) [inline TempoWorld/Source/TempoDebris/Public/PCGAssignRandomMeshAttribute.h:46]: Say which elements this one is similar to

File: TempoWorld/Source/TempoDebris/Private/PCGAssignRandomMeshAttribute.cpp [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+	}
+
+	FStaticMeshDistribution* Row = DataTable->FindRow<FStaticMeshDistribution>(RowName, TEXT(""), false);
+	if (!Row)
+	{
+		PCGE_LOG(Error, GraphAndLog, FText::Format(LOCTEXT("NoRowFound", "FPCGAssignRandomMeshAttribute: Data table '{0}' has no row named '{1}'"), FText::FromString(DataTable->GetPathName()), FText::FromName(RowName)));
+		return true;
+	}
+
+	TMap<TSoftObjectPtr<UStaticMesh>, float> MeshDistribution;
```
Comment (pete-tempo) [inline TempoWorld/Source/TempoDebris/Private/PCGAssignRandomMeshAttribute.cpp:139]: Sort the map

**Review by gio-tempo — APPROVED** [review]
Reviewed over VC.

File: TempoWorld/Source/TempoDebris/Private/PCGAssignRandomMeshAttribute.cpp (hunk 3 above)
Comment (pete-tempo) [reply TempoWorld/Source/TempoDebris/Private/PCGAssignRandomMeshAttribute.cpp:139]: I actually tested it and I *haven't* found any evidence that TMap iteration isn't deterministic:
https://github.com/tempo-sim/Tempo/tree/map_determinism_test

Still, I'm leery. So I'm sorting it.

**Pull request merged**